src/Microsoft.DotNet.Build.Tasks.Packaging/src/GetLayoutFiles.cs

```diff
@@ -0,0 +1,139 @@
+
+namespace Microsoft.DotNet.Build.Tasks.Packaging
+{
+    public class GetLayoutFiles : PackagingTask
+    {
+        /// <summary>
+        /// Package report files
+        /// </summary>
+        [Required]
+        public string[] PackageReports { get; set; }
+
+        /// <summary>
+        /// Destination directory for layout
+        /// </summary>
+        [Required]
+        public string DestinationDirectory { get; set; }
+
+        /// <summary>
+        /// Optional set of frameworks to restrict the layout
+        ///   Identity: Framework
+        ///   RuntimeIDs: Semi-colon seperated list of runtime IDs
+        /// </summary>
+        public ITaskItem[] Frameworks { get; set; }
+
+        /// <summary>
+        /// Files to be copied to layout
+        ///   Identity : source path
+        ///   Destination : destination path
+        /// </summary>
+        [Output]
+        public ITaskItem[] LayoutFiles { get; set; }
+        
+        public override bool Execute()
+        {
+            var frameworks = Frameworks.NullAsEmpty().ToDictionary(
+                i => NuGetFramework.Parse(i.ItemSpec),
+                i => 
+                {
+                    var rids = i.GetMetadata("RuntimeIds");
+                    return String.IsNullOrEmpty(rids) ? new HashSet<string>() : new HashSet<string>(rids.Split(';'));
+                },
+                NuGetFramework.Comparer);
+
+            var layoutFiles = new List<ITaskItem>();
+
+            foreach(var packageReportFile in PackageReports)
+            {
+                var packageReport = PackageReport.Load(packageReportFile);
+
+                foreach(var targetInfo in packageReport.Targets)
+                {
+                    var targetName = targetInfo.Key;
+                    var target = targetInfo.Value;
+
+                    var targetParts = targetName.Split('/');
+
+                    var fx = NuGetFramework.Parse(targetParts[0]);
+                    string rid = null;
+
+                    if (targetParts.Length > 1)
+                    {
+                        rid = targetParts[1];
+                    }
+                    
+                    if (frameworks.Count != 0)
+                    {
+                        HashSet<string> rids = null;
+
+                        if (!frameworks.TryGetValue(fx, out rids))
+                        {
+                            Log.LogMessage(LogImportance.Low, $"Skipping {fx} since it is not in {nameof(Frameworks)}");
+                            continue;
+                        }
+
+                        if (rid != null && rids.Count > 0 && !rids.Contains(rid))
+                        {
+                            Log.LogMessage(LogImportance.Low, $"Skipping {fx}/{rid} since it is not in {nameof(Frameworks)}");
+                            continue;
+                        }
+                    }
+
+                    if (!packageReport.SupportedFrameworks.ContainsKey(fx.ToString()))
```

Comment:
I list all targets even those that might not be supported since that can help folks debug what we consider a package as not supported.